src/components/Calculator.js

```diff
@@ -1,41 +1,101 @@
-import React from 'react';
+import React, { useState } from 'react';
+import calculate from '../logic/calculate';
 import './Calculator.css';
 
 // eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
-  render() {
-    return (
-      <div className="calculator">
-        <div className="result">
-          <p>0</p>
-        </div>
-
-        <span>AC</span>
-        <span>+/-</span>
-        <span>%</span>
-        <span className="operator">÷</span>
-
-        <span>7</span>
-        <span>8</span>
-        <span>9</span>
-        <span className="operator">x</span>
-
-        <span>4</span>
-        <span>5</span>
-        <span>6</span>
-        <span className="operator">-</span>
-
-        <span>1</span>
-        <span>2</span>
-        <span>3</span>
-        <span className="operator">+</span>
-
-        <span className="double">0</span>
-        <span>.</span>
-        <span className="operator">=</span>
-      </div>
-    );
+// eslint-disable-line jsx-a11y/no-static-element-interactions
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const updateObject = (e) => {
+    const buttonName = e.target.textContent;
+    const myObj = calculate(state, buttonName);
+    setState(myObj);
+    // console.log(myObj);
+  };
+
+  const { total, next, operation } = state;
+  const op = operation === '%' ? 'mod' : operation;
```

Comment:
- [ ] Kindly make sure to display the mod sign when we calculate a value with mod %.
Hint: replace mod with % in order to display that.

https://user-images.githubusercontent.com/70269507/188856237-031c74ca-17b3-48d5-83e2-4d262769e8f4.mov